apis/appmesh/v1beta2/virtualgateway_types.go

```diff
@@ -17,25 +17,262 @@ limitations under the License.
+type VirtualGatewayPortMapping struct {
+	// The port used for the port mapping.
+	Port PortNumber `json:"port"`
+	// The protocol used for the port mapping.
+	Protocol VirtualGatewayPortProtocol `json:"protocol"`
+}
+
+type VirtualGatewayCondition struct {
+	// Type of VirtualGateway condition.
+	Type VirtualGatewayConditionType `json:"type"`
+	// Status of the condition, one of True, False, Unknown.
+	Status corev1.ConditionStatus `json:"status"`
+	// Last time the condition transitioned from one status to another.
+	// +optional
+	LastTransitionTime *metav1.Time `json:"lastTransitionTime,omitempty"`
+	// The reason for the condition's last transition.
+	// +optional
+	Reason *string `json:"reason,omitempty"`
+	// A human readable message indicating details about the transition.
+	// +optional
+	Message *string `json:"message,omitempty"`
+}
+
+// VirtualGatewayHealthCheckPolicy refers to https://docs.aws.amazon.com/app-mesh/latest/APIReference/API_HealthCheckPolicy.html
+type VirtualGatewayHealthCheckPolicy struct {
+	// The number of consecutive successful health checks that must occur before declaring listener healthy.
+	// If unspecified, defaults to be 10
+	// +kubebuilder:validation:Minimum=2
+	// +kubebuilder:validation:Maximum=10
+	// +optional
+	HealthyThreshold *int64 `json:"healthyThreshold,omitempty"`
+	// The time period in milliseconds between each health check execution.
+	// If unspecified, defaults to be 30000
+	// +kubebuilder:validation:Minimum=5000
+	// +kubebuilder:validation:Maximum=300000
+	// +optional
+	IntervalMillis *int64 `json:"intervalMillis,omitempty"`
+	// The destination path for the health check request.
+	// This value is only used if the specified protocol is http or http2. For any other protocol, this value is ignored.
+	// +optional
+	Path *string `json:"path,omitempty"`
+	// The destination port for the health check request.
+	// If unspecified, defaults to be same as port defined in the PortMapping for the listener.
+	// +optional
+	Port *PortNumber `json:"port,omitempty"`
+	// The protocol for the health check request
+	// If unspecified, defaults to be same as protocol defined in the PortMapping for the listener.
+	// +optional
+	Protocol *VirtualGatewayPortProtocol `json:"protocol,omitempty"`
+	// The amount of time to wait when receiving a response from the health check, in milliseconds.
+	// If unspecified, defaults to be 5000
+	// +kubebuilder:validation:Minimum=2000
+	// +kubebuilder:validation:Maximum=60000
+	// +optional
+	TimeoutMillis *int64 `json:"timeoutMillis,omitempty"`
+	// The number of consecutive failed health checks that must occur before declaring a virtual Gateway unhealthy.
+	// If unspecified, defaults to be 2
+	// +kubebuilder:validation:Minimum=2
+	// +kubebuilder:validation:Maximum=10
+	// +optional
+	UnhealthyThreshold *int64 `json:"unhealthyThreshold,omitempty"`
```

Comment:
From the model [here](https://raw.githubusercontent.com/aws/aws-app-mesh-roadmap/master/appmesh-preview/service-model.json) following should be required fields:
```
"HealthCheckPolicy": {
      "type": "structure",
      "required": [
        "healthyThreshold",
        "intervalMillis",
        "protocol",
        "timeoutMillis",
        "unhealthyThreshold"
      ]

```

Reply:
These values have defaults if not specified in Virtual Node. Hence I followed the same convention.